Issues #4 & #5: enable additional payload fields (client_uri, client_name, etc.)

jahabeebs opened the pull request:
This PR closes #14 
This PR closes #15 

### Description
- Added backend logic to allow for updating client_uri, client_name, tos_uri, logo_uri, policy_uri
- Added frontend forms to allow for updating the above values via the UI and added error toast popup on failure
- Pushed separate commit with lint fixes

### Screenshots
Filling in form info 

<img width="1051" alt="filling in info" src="https://user-images.githubusercontent.com/47253537/213300109-cb895d96-e299-4bdd-9b1b-e88eda89b818.png">

Successful update

<img width="1375" alt="updated" src="https://user-images.githubusercontent.com/47253537/213300234-15f6e820-7480-4ce5-b18c-e935eb96617a.png">

Failure toast message

<img width="1398" alt="failure" src="https://user-images.githubusercontent.com/47253537/213300275-056ede50-7eb8-494d-8912-3c2f58cefd1c.png">


### Checklist
- [X] Fix all lint + build issues with CI
- [X] Request a review

pi0neerpat commented on api/src/graphql/oidc.sdl.js:
```diff
@@ -9,7 +9,15 @@ export const schema = gql`
+    updateClient(
+      id: String!
+      redirectUrls: String!
+      clientURI: String!
+      clientName: String!
+      tosURI: String!
+      logoURI: String!
+      policyURI: String!
```
I don't think these should always be required, in case the developer doesn't have a link yet

pi0neerpat commented on api/src/services/clients/clients.js:
```diff
@@ -60,7 +68,17 @@ export const updateClient = async ({ id, redirectUrls }) => {
+    data: {
+      payload: {
+        ...client.payload,
+        client_uri: clientURI,
+        client_name: clientName,
+        tos_uri: tosURI,
+        logo_uri: logoURI,
+        policy_uri: policyURI,
+        redirect_uris: list,
```
If the element is empty, don't update it eg

```
...(tosURI && { tosURI} )
```

jahabeebs commented:
@pi0neerpat 
- Added null checks for clientURI, clientName, tosURI, logoURI, policyURI in updateClient
- Removed client-side validation in the form for the above fields
- Removed the `!` non-null validation for above fields in `schema.prisma`, `oidc.sdl.js`, and `OidcCell.js`

Blank update does not change fields

https://user-images.githubusercontent.com/47253537/213535522-6c08c30a-3514-41da-bc82-ad2285bfcb2a.mov

Successful update

https://user-images.githubusercontent.com/47253537/213535582-22745c2f-e399-4c26-9963-1beb4cf08cc0.mov

redirectURLs is still validated 

<img width="578" alt="validation redirectURLs" src="https://user-images.githubusercontent.com/47253537/213535615-3dcc7922-edd3-461d-8cce-dc16c4bbc37a.png">

pi0neerpat reviewed (approved):
:+1:
